src/lib/pages/backoffice/Actions/CheckIn/ItemsSearch/ItemsSearch.js

```diff
@@ -7,55 +7,19 @@ import _isEmpty from 'lodash/isEmpty';
 import { Header } from 'semantic-ui-react';
 
 export default class ItemsSearch extends Component {
-  constructor(props) {
-    super(props);
-    this.state = { queryString: '' };
-  }
-
-  onInputChange = queryString => {
-    this.setState({
-      queryString: queryString,
-    });
-  };
-
   executeCheckinAndClearInput = async queryString => {
     const { checkin } = this.props;
     if (queryString.trim() === '') return;
     await checkin(queryString);
-    this.setState({ queryString: '' });
-  };
-
-  onPasteHandler = event => {
-    const queryString = (event.clipboardData || window.clipboardData).getData(
-      'text'
-    );
-    this.executeCheckinAndClearInput(queryString);
-  };
-
-  onKeyPressHandler = event => {
-    const { queryString } = this.state;
-    if (event.key === 'Enter') {
-      this.executeCheckinAndClearInput(queryString);
-    }
   };
 
   render() {
     const { loans } = this.props;
-    const { queryString } = this.state;
     return (
       <>
-        <SearchBar
-          action={{
-            icon: 'search',
-            onClick: () => this.executeCheckinAndClearInput(queryString),
-          }}
-          queryString={queryString}
-          updateQueryString={this.executeCheckinAndClearInput}
+        <SearchBarILS
+          onSearchHandler={this.executeCheckinAndClearInput}
```

Comment:
are we losing the `onPaste` and the `onKeyPressHandler` `ENTER` here?
Same for the others below

Reply:
By default now, on every action `click`, `paste` and `key` (only if it is enter), we execute search.